### PR TITLE
Optimize readme publish workflow

### PR DESCRIPTION
**Pull request by jitendra42**
## Changes Made

- 
- [x] The code follows the project's [coding standards](https://github.com/intel/ai-containers/blob/main/CONTRIBUTING.md#code-style).
- [x] No Intel Internal IP is present within the changes.
- [x] The documentation has been updated to reflect any changes in functionality.

## Validation

- [ ] I have tested any changes in container groups locally with [`test_runner.py`](https://github.com/intel/ai-containers/blob/main/test-runner/README.md) with all existing tests passing, and I have added new tests where applicable.

### CODE REVIEW

File: .github/workflows/dockerhub-description.yml [hunk 1]
```diff
@@ -23,17 +23,27 @@ jobs:
     outputs:
       matrix: ${{ steps.set-matrix.outputs.matrix }}
     steps:
-    - name: Harden Runner
-      uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7 # v2.10.1
-      with:
-        egress-policy: audit
```
Comment (tylertitsworth) [inline .github/workflows/dockerhub-description.yml:29]: Why are we removing this?

File: .github/workflows/dockerhub-description.yml [hunk 2]
```diff
@@ -23,17 +23,27 @@ jobs:
...
+        # Get the list of files changed in the latest commit(s)
+        changed_files=$(git diff --name-only HEAD~1 ${{ github.sha }} | sed 's|^|./|' | jq -R . | jq -s .)
+        echo "Changed files: $changed_files"
+
+        # If there are changed files, filter the JSON using jq
+        matrix=$(jq -c --argjson changed "$changed_files" \
+          '.readmes |= map(select(.fname as $fname | any($changed[]; . == $fname)))' \
+          .github/dockerhub-readmes.json)
+        echo "Filtered matrix:"
+        echo $matrix
```
Comment (tylertitsworth) [inline .github/workflows/dockerhub-description.yml:41]: Output matrix will be printed in debug mode, these echo's may not be necessary for prod version

File: .github/workflows/dockerhub-description.yml [hunk 3]
```diff
@@ -23,17 +23,27 @@ jobs:
...
+        echo "matrix=$matrix" >> $GITHUB_OUTPUT
   publish-dockerhub-description:
     runs-on: ${{ github.repository_owner == 'intel' && 'intel-ubuntu-latest' || 'ubuntu-latest' }}
     needs: setup-matrix
+    if: ${{ needs.setup-matrix.outputs.matrix != '{"readmes":[]}' }}
```
Comment (tylertitsworth) [inline .github/workflows/dockerhub-description.yml:46]: Could you not just do `matrix.readmes`?

**Review by tylertitsworth — CHANGES_REQUESTED** [review]
Small nits

File: .github/workflows/dockerhub-description.yml (hunk 2 above)
Comment (jitendra42) [reply .github/workflows/dockerhub-description.yml]: removed

File: .github/workflows/dockerhub-description.yml (hunk 3 above)
Comment (jitendra42) [reply .github/workflows/dockerhub-description.yml:48]: if condition needs to be evaluated on matrix, otherwise it errors out.

**Comment by jitendra42** [comment]
tests
no files change: https://github.com/jitendra42/ai-containers/actions/runs/11523502805
file changes: https://github.com/jitendra42/ai-containers/actions/runs/11523508399 ( it fails on authentication as I dont have dockerhub secrets, which is expected

**Pull request merged**